src/service/repo/signedMessageRepo.ts

```diff
@@ -189,6 +200,34 @@ async function getSignedChannelDetail(channelType: GetstreamChannelType, channel
   }
 }
 
+async function changeSignedChannelDetail(
+  channelId: string,
+  channelName?: string,
+  channelImage?: string
+) {
+  const payload: ChangeGroupInfoPayload = {
+    channel_id: channelId
+  };
+  if (channelName) {
+    payload.channel_name = channelName;
+  }
+  if (channelImage) {
+    payload.channel_image = channelImage;
+  }
+
+  try {
+    const response = await api.post(baseUrl.changeSignedChannelDetail, payload);
+    if (response.status === 200) {
+      return Promise.resolve(response.data?.data);
+    }
+
+    return Promise.reject(response.data?.data);
+  } catch (e) {
+    console.log(e?.response);
+    return Promise.reject(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in this function could be improved. Currently, the error is just logged to the console and then rejected. It would be better to provide a more descriptive error message when rejecting the promise. This will make it easier for the calling code to handle the error appropriately.
```diff
-    console.log(e?.response);
-    return Promise.reject(e);
+    console.error('Error changing signed channel detail:', e?.response || e);
+    return Promise.reject(new Error('Failed to change signed channel detail'));
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->